src/twisted/mail/relaymanager.py

```diff
@@ -1096,9 +1100,9 @@ def _cbMX(self, answers, domain, cnamesLeft):
                 exchanges.append((record.preference, record))
 
         if exchanges:
-            exchanges.sort()
+            exchanges = sorted(exchanges, key=lambda k: str(k))
```

Comment:
This seems incorrect: we should sort numerically on the preference field, not lexicographically on the repr. Maybe `key=lambda k: k[0]` ?

Is there a reason to use `sorted()` instead of sorting in-place?